lib/Backend/GlobOpt.cpp

```diff
@@ -7184,13 +7212,29 @@ GlobOpt::SetSymStoreDirect(ValueInfo * valueInfo, Sym * sym)
     if (prevSymStore && prevSymStore->IsStackSym() &&
         prevSymStore->AsStackSym()->HasByteCodeRegSlot())
     {
-        Assert(this->blockData.changedSyms != nullptr);
-        this->blockData.changedSyms->Set(prevSymStore->m_id);
+        this->SetChangedSym(prevSymStore->m_id);
     }
     valueInfo->SetSymStore(sym);
 }
 
 void
+GlobOpt::SetChangedSym(SymID symId)
+{
+    GlobOptBlockData * globOptData = &this->currentBlock->globOptData;
```

Comment:
It's not just MergeCaptureValues though.  The invariant code/PRE and others seem to call SetSymStore for values not in the current block.